cpp/src/parquet/statistics.cc

```diff
@@ -53,6 +55,25 @@ namespace {
 constexpr int value_length(int value_length, const ByteArray& value) { return value.len; }
 constexpr int value_length(int type_length, const FLBA& value) { return type_length; }
 
+// Static "constants" for normalizing float16 min/max values. These need to be expressed
+// as pointers because `Float16LogicalType` represents an FLBA.
+const uint8_t* float16_lowest() {
+  static const auto bytes = std::numeric_limits<Float16>::lowest().ToLittleEndian();
```

Comment:
This could probably made `constexpr` by spelling out the `#if ARROW_LITTLE_ENDIAN` dance explicitly in `ToLittleEndian` and friends.
Not sure that's useful here, but worth remembering.